src/components/Integration.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <section>
+    <h3 v-text="description" />
+    <section class="body">
+      <Banner v-if="globalError" color="error" :label="globalError" />
+      <ul>
+        <li v-for="item of integrationsList" :key="item.name">
+          <checkbox
+            :value="item.value"
+            :label="item.name"
+            :disabled="item.disabled"
+            @input="toggleIntegration(item.name, $event)"
+          />
+        </li>
+      </ul>
+    </section>
+  </section>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import type { PropType } from 'vue';
+import Component from 'vue-class-component';
+import Banner from '@/components/Banner.vue';
+import Card from '@/components/Card.vue';
+import Checkbox from '@/components/form/Checkbox.vue';
+import paths from '@/utils/paths';
+
+const IntegrationProps = Vue.extend({
+  props: {
+    title: {
+      type:    String,
+      default: 'System Integration',
+    },
+    description: {
+      type:    String,
+      default: '',
+    },
+    integrations: {
+      type:    Object as PropType<Record<string, boolean | string>>,
+      default: () => ({}),
+    },
+  },
+});
+
+@Component({
+  components: {
+    Banner, Card, Checkbox
+  }
+})
+class Integration extends IntegrationProps {
+  /**
+   * A mapping to temporarily disable a selection while work happens
+   * asynchronously, to prevent the user from retrying to toggle too quickly.
+   */
+  protected busy: Record<string, boolean> = {};
+  protected GlobalFailureIntegrationName = paths.integration;
```

Comment:
https://github.com/rancher-sandbox/rancher-desktop/blob/732e876b3dc66c571e62688455eead18f8a61fc4/src/utils/paths.ts#L92-L94

This throws an exception. It doesn't make sense on Windows anyway (we're not doing tools here).